### PR TITLE
New handling of BGC prediction

### PR DESCRIPTION
**Pull request by alexeigurevich**
This is the first (quite dirty) implementation of the new interface (`src/data_types.py`) for BGC variants. 
The idea is to still rely on the intermediate text files (antiSMASH v3-like, e.g. `<>_nrpspredictor2_codes.txt`) but parse them into new data types. In the future, the files will be omitted and we will parse the content of the BGC variants directly from the antiSMASH JSON file.

In the current implementation, the residue scores are filled simply from the old scores (ranging from 0 to 100). I will add further commits to adjust the scoring to the new model (getting `aa10_score`, `aa34_score`, `svm_small_cluster`, etc for every Nerpa-supported residue (out of 60)). This will basically require the update of `src/nerpa_pipeline/NRPSPredictor_utils/codes_handler.py` and more specifically `def get_prediction_from_signature` and a few functions around.

After that @azmtag can add his classifier on top of it and we would be able to merge the fully functional pull request.

### CODE REVIEW

File: nerpa.py [hunk 1]
```diff
@@ -284,7 +284,8 @@ def run(args, log):
     log.start()
 
     if args.predictions is not None:
-        path_predictions = copy_prediction_list(args, output_dir)
+        raise NotImplemented('Precomputed BGC predictions not supported yet')
```
Comment (ilianolhin) [inline nerpa.py]: Did we ever (and do we have any plan to) support precomputed BGC predictions?

Comment (alexeigurevich) [reply nerpa.py]: Yes, we did, and probably it makes sense to have. If a user wishes to rerun the analysis of a huge number of antiSMASH predictions vs the new set of structures, s/he can just reuse already generated BGC variants. Now it should be super easy with just YAML dump/load functions.

**Comment by alexeigurevich** [comment]
I pushed the new scoring with two stubs for @azmtag (look for "TODO: Azat:"). Currently, everything works with a dummy model that simply reports `log(aa10_score)` for all Nerpa-supported residues. 
Note: I excluded "LDAP" residue completely since it turned out to be a synonym for "drp" which was already there. Now we have 58 residues (and "unknown")

**Comment by azmtag** [comment]
- Ilia's conda env + my changes. For now, we need it only because sklearn & joblib versions should match the one I used for saving the model
- the model is integrated into the code
- a few as7 outputs for testing

**Comment by ilianolhin** [comment]
Lesha, could you remove writing BGC variants in the output folder? I would prefer to write everything in the end in `write_results` function

**Comment by azmtag** [comment]
I'd suggest postponing it, as all this output could be useful for debugging

**Comment by alexeigurevich** [comment]
Ok, I added the `--debug` option to print these YAMLs (and control similar things in the future). By default, dumping will be skipped. I hope this solution satisfies everyone :)

**Comment by alexeigurevich** [comment]
@azmtag is it possible to reduce the conda environment file to only essential dependencies? I had troubles installing it on my laptop. At the same time, installing `scikit-learn`, `joblib` (as well as `pandas` and `pyyaml`) even in Python 3.10 env did the trick for me.

**Comment by azmtag** [comment]
> @azmtag is it possible to reduce the conda environment file to only essential dependencies? I had troubles installing it on my laptop. At the same time, installing `scikit-learn`, `joblib` (as well as `pandas` and `pyyaml`) even in Python 3.10 env did the trick for me.

Yep, this file is basically the output of mamba env export, so all packages are pinned to versions specific to linux64. Currently it's just for reference, we can do a proper one later

**Pull request merged**